tart2ms/tart2ms.py

```diff
@@ -690,43 +778,147 @@ def __twelveball(direction):
-            LOGGER.info(
-                f"Per user request: Rephase all data to {new_phase_dir_repr}")
-            rephased_data = da.empty_like(dask_data)
-            rephased_data = \
-                da.map_blocks(rephase,
-                              dask_data,
-                              dtype=dask_data.dtype,
-                              chunks=dask_data.chunks,
-                              # kwargs for rephase
-                              freq=spw_chan_freqs[spw_id],
-                              pos=np.rad2deg(centroid_direction[0, :]),
-                              uvw=uvw_data,
-                              refdir=np.rad2deg(zenith_directions),
-                              field_ids=map_row_to_zendir)
-            dask_data = rephased_data
-            if fill_model and sources:
+            if model_data is None:
+                model_data = da.zeros_like(dask_data)
+            cat_sources = get_catalog_sources_azel(obstime, location)
+            if write_extragalactic_catalogs:
+                LOGGER.info(f"Predicting celestial catalog positions for {len(epoch_s)} timestamps")
+                celestial_model = predict_model(dask_data.shape, dask_data.chunks, dask_data.dtype, 
+                                            uvw_data,
+                                            epoch_s, spw_chan_freqs, spw_i,
+                                            zenith_directions,
+                                            map_row_to_zendir,
+                                            location,
+                                            cat_sources, epoch_s, obstime,
+                                            writemodelcatalog,
+                                            filter_elevation=20.0,
+                                            append_catalog=True)
+                if celestial_model is None:
+                    celestial_model = da.zeros_like(dask_data)
+                model_data += celestial_model
+            LOGGER.info(f"Predicting Sun and Moon positions for {len(epoch_s)} timestamps")
+            cat_sources = get_solar_system_bodies(obstime, location)
+            solar_model = predict_model(dask_data.shape, dask_data.chunks, dask_data.dtype, 
+                                        uvw_data,
+                                        epoch_s, spw_chan_freqs, spw_i,
+                                        zenith_directions,
+                                        map_row_to_zendir,
+                                        location,
+                                        cat_sources, epoch_s, obstime,
+                                        writemodelcatalog,
+                                        filter_elevation=20.0,
+                                        append_catalog=True)
+
+            if solar_model is None:
+                solar_model = da.zeros_like(dask_data)
+            model_data += solar_model
+            model_data.rechunk(dask_data.chunks)
```

Comment:
We augment GNSS positions with extragalactic quasars and fluxy black bodies here. Can probably add Mars etc at some point